point/gemini_commands.py

```diff
@@ -696,32 +723,81 @@ def __init__(self, ra):
     def response(self):  return G2Rsp_SetObjectRA(self)
 class G2Rsp_SetObjectRA(Gemini2Response_LX200_FixedLength):
     def fixed_len(self): return 1
     def interpret(self):
-        validity = G2Valid(self.get_raw())  # raises ValueError if the response field value isn't in the enum
-        if validity != G2Valid.VALID: raise G2ResponseInterpretationError()
+        validity = G2Valid(self.get_raw()) # raises ValueError if the response field value isn't in the enum
+        if validity != G2Valid.VALID: raise G2ResponseInterpretationFailure()
 
 class G2Cmd_SetObjectDec(Gemini2Command_LX200):
     def __init__(self, dec):
         if dec < -90.0 or dec > 90.0:
             raise G2CommandParameterValueError('dec must be >= -90.0 and <= 90.0')
         sign, self._deg, self._min, self._sec = ang_to_degminsec(dec)
         self._signchar = '+' if sign >= 0.0 else '-'
-    def lx200_str(self): return 'Sd{:s}{:02d}:{:02d}:{:02d}'.format(self._signchar, self._deg, self._min, int(self._sec))
+    def lx200_str(self): return 'Sd{:s}{:02d}:{:02d}:{:02d}'.format(self._signchar, self._deg, self._min, self._sec)
     def response(self):  return G2Rsp_SetObjectDec(self)
 class G2Rsp_SetObjectDec(Gemini2Response_LX200_FixedLength):
     def fixed_len(self): return 1
     def interpret(self):
-        validity = G2Valid(self.get_raw())  # raises ValueError if the response field value isn't in the enum
-        if validity != G2Valid.VALID: raise G2ResponseInterpretationError()
+        validity = G2Valid(self.get_raw()) # raises ValueError if the response field value isn't in the enum
+        if validity != G2Valid.VALID: raise G2ResponseInterpretationFailure()
         # NOTE: only objects which are currently above the horizon are considered valid
 
+class G2Cmd_SetSiteLongitude(Gemini2Command_LX200):
+    def __init__(self, lon):
+        if lon <= -360.0 or lon >= 360.0:
+            raise G2CommandParameterValueError('lon must be > -360.0 and < 360.0')
+        sign, self._deg, self._min, = ang_to_degmin(lon)
+        # everyone else in the world uses positive to mean eastern longitudes; but not LX200!
+        self._signchar = '-' if sign >= 0.0 else '+'
+    def lx200_str(self): return 'Sg{:s}{:03d}*{:02d}'.format(self._signchar, self._deg, self._min)
+    def response(self):  return G2Rsp_SetSiteLongitude(self)
+class G2Rsp_SetSiteLongitude(Gemini2Response_LX200_FixedLengthOrZero):
+    def fixed_len(self): return 1
+    def interpret(self):
+        if len(self.get_raw()) == 0: raise G2ResponseInterpretationFailure() # invalid
+        if self.get_raw() != '1':    raise G2ResponseInterpretationFailure() # ???
+
+class G2Cmd_SetSiteLatitude(Gemini2Command_LX200):
+    def __init__(self, lat):
+        if lat < -90.0 or lat > 90.0:
+            raise G2CommandParameterValueError('lat must be >= -90.0 and <= 90.0')
+        sign, self._deg, self._min, _ = ang_to_degmin(lat)
```

Comment:
Too many return values expected for `ang_to_degmin`

Reply:
Good catch, will fix